packages/mobile/src/onboarding/contacts/ImportContactsScreen.tsx

```diff
@@ -1,227 +1,187 @@
+  const importStatus = importContactsProgress.status
+  const prevImportContactProgressRef = React.useRef(importContactsProgress)
+  const matchedContacts = useSelector(matchedContactsSelector)
+  const matchedContactsCount = Object.keys(matchedContacts).length
+  const dispatch = useDispatch()
 
-const mapDispatchToProps = {
-  importContacts,
-  cancelImportContacts,
-  denyImportContacts,
-}
-
-class ImportContactScreen extends React.Component<Props, State> {
-  static navigationOptions = nuxNavigationOptions
-
-  state: State = {
-    isFindMeSwitchChecked: true,
-  }
-
-  onFinishTimeout: number | null = null
-
-  componentDidUpdate(prevProps: Props) {
-    if (
-      prevProps.importContactsProgress.status !== ImportContactsStatus.Done &&
-      this.props.importContactsProgress.status === ImportContactsStatus.Done
-    ) {
-      // Set timeout to leave checkmark in place for a little while
-      // @ts-ignore setTimeout is incorrectly picking up the Node typings instead of RN's typings
-      this.onFinishTimeout = setTimeout(this.onFinish, 1500)
-    }
-  }
-
-  componentWillUnmount() {
-    if (this.onFinishTimeout) {
-      clearTimeout(this.onFinishTimeout)
-    }
-  }
-
-  onPressConnect = async () => {
+  async function onPressConnect() {
     CeloAnalytics.track(CustomEventNames.import_contacts)
     const hasGivenContactPermission = await requestContactsPermission()
     if (hasGivenContactPermission) {
-      this.props.importContacts(this.state.isFindMeSwitchChecked)
+      dispatch(importContacts(isFindMeSwitchChecked))
     }
   }
 
-  onPressSkip = () => {
-    this.props.cancelImportContacts()
-    this.props.denyImportContacts()
-    CeloAnalytics.track(CustomEventNames.import_contacts_skip)
-    this.onFinish()
-  }
-
-  onToggleFindMeSwitch = (value: boolean) => {
-    this.setState({
-      isFindMeSwitchChecked: value,
-    })
-  }
-
-  onFinish = () => {
+  function onFinish() {
     navigate(Screens.OnboardingSuccessScreen)
   }
 
-  renderImportStatus = () => {
-    const {
-      t,
-      importContactsProgress: { status, total },
-      matchedContacts,
-    } = this.props
-    const matchesFound = Object.keys(matchedContacts).length
+  function onToggleFindMeSwitch(value: boolean) {
+    setFindMeSwitch(value)
+  }
 
-    if (status === ImportContactsStatus.Done) {
-      return (
-        <View style={styles.statusContainer}>
-          <Checkmark />
-          {matchesFound > 0 ? (
-            <>
-              <Text style={styles.h1Status}>
-                {t('contacts.syncing.successHeader', { matches: matchesFound })}
-              </Text>
-              <Text style={styles.h2Status}>{t('contacts.syncing.successStatus')}</Text>
-            </>
-          ) : (
-            <Text style={styles.h1Status}>{t('contacts.syncing.successStatus')}</Text>
-          )}
-        </View>
-      )
+  // If import has already been done, move user forward
```

Comment:
When would this happen?

Reply:
Unclear if it's possible in the wild. It was happening pretty frequently while testing so I built it as a catch-all. If you can't think of any instance where this would happen (I can't off top of my head), I am happy to remove it

Reply:
If it happened, it seems like a bug that we didn't correctly reset the import status before coming here.
It shouldn't ever happen cause you need to get verified before we import contacts? Maybe one edge case is the invite screen? We should just cut that crappy screen IMO haha

Reply:
I was bumping into it when using the dev verification toggle, which we don't need to worry about in prod.

I think all routes to the Invite screen are being cut off as part of the redesign so I will go ahead and remove this catch-all then.